src/mjolnir/pbfgraphparser.cc

```diff
@@ -855,8 +855,8 @@ struct graph_callback : public OSMPBF::Callback {
     // if this nodes id is less than the waynode we are looking for then we know its a node we can
     // skip because it means there were no ways that we kept that referenced it. also we could run out
     // of waynodes to look for and in that case we are done as well
-    if (osmid < (*(*way_nodes_)[current_way_node_index_]).node.osmid_ ||
-        current_way_node_index_ >= way_nodes_->size()) {
+    if (current_way_node_index_ >= way_nodes_->size() ||
```

Comment:
Should this be `>`, not `>=` ?  If `current_way_node_index_` == `way_nodes_->size()`, then we'll access 1 past the end of the array, even with your fix.

Reply:
if `current_way_node_index_  == way_nodes_->size()` then it immediately jump to `return;` without checking the second condition, nope?

Reply:
🤦 I had it backwards, nevermind!